src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -58,14 +58,26 @@ static ucs_status_t uct_cuda_copy_iface_get_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
-static int uct_cuda_copy_iface_is_reachable(const uct_iface_h tl_iface,
-                                            const uct_device_addr_t *dev_addr,
-                                            const uct_iface_addr_t *iface_addr)
+static int uct_cuda_copy_iface_is_reachable_v2(
+        const uct_iface_h tl_iface,
+        const uct_iface_is_reachable_params_t *params)
 {
-    uct_cuda_copy_iface_t  *iface = ucs_derived_of(tl_iface, uct_cuda_copy_iface_t);
-    uct_cuda_copy_iface_addr_t *addr = (uct_cuda_copy_iface_addr_t*)iface_addr;
+    uct_cuda_copy_iface_t *iface = ucs_derived_of(tl_iface,
+                                                  uct_cuda_copy_iface_t);
+    uct_cuda_copy_iface_addr_t *addr;
 
-    return (addr != NULL) && (iface->id == *addr);
+    if (!uct_iface_is_reachable_params_valid(
+                params, UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR |
+                                UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR)) {
```

Comment:
minor: can be a short separate function to simplify the code (as it is used in many places)

Reply:
fixed